lib/inets/src/http_client/httpc.erl

```diff
@@ -861,26 +863,41 @@ mk_chunkify_fun(ProcessBody) ->
     end.
 
 
-handle_answer(RequestId, false, _) ->
+handle_answer(RequestId, _, false, _, _) ->
     {ok, RequestId};
-handle_answer(RequestId, true, Options) ->
+handle_answer(RequestId, ClientAlias, true, Options, Timeout) ->
     receive
-	{http, {RequestId, saved_to_file}} ->
-	    {ok, saved_to_file};
-	{http, {RequestId, {_,_,_} = Result}} ->
-	    return_answer(Options, Result);
-	{http, {RequestId, {error, Reason}}} ->
-	    {error, Reason}
-    end.
-
-return_answer(Options, {StatusLine, Headers, BinBody}) ->
-    Body = maybe_format_body(BinBody, Options),
-    case proplists:get_value(full_result, Options, true) of
-	true ->
-	    {ok, {StatusLine, Headers, Body}};
-	false ->
-	    {_, Status, _} = StatusLine,
-	    {ok, {Status, Body}}
+        {http, {RequestId, {ok, saved_to_file}}} ->
+            unalias(ClientAlias),
+            {ok, saved_to_file};
+        {http, {RequestId, {error, Reason}}} ->
+            unalias(ClientAlias),
+            {error, Reason};
+        {http, {RequestId, {ok, {StatusLine,Headers,BinBody}}}} ->
+            unalias(ClientAlias),
+            Body = maybe_format_body(BinBody, Options),
+            {ok, {StatusLine, Headers, Body}};
+        {http, {RequestId, {ok, {StatusCode,BinBody}}}} ->
+            unalias(ClientAlias),
+            Body = maybe_format_body(BinBody, Options),
+            {ok, {StatusCode, Body}}
```

Comment:
```suggestion
        {http, {RequestId, {ok, saved_to_file}}} ->
            true = unalias(ClientAlias),
            {ok, saved_to_file};
        {http, {RequestId, {error, Reason}}} ->
            true = unalias(ClientAlias),
            {error, Reason};
        {http, {RequestId, {ok, {StatusLine, Headers, BinBody}}}} ->
            true = unalias(ClientAlias),
            Body = maybe_format_body(BinBody, Options),
            {ok, {StatusLine, Headers, Body}};
        {http, {RequestId, {ok, {StatusCode, BinBody}}}} ->
            true = unalias(ClientAlias),
            Body = maybe_format_body(BinBody, Options),
            {ok, {StatusCode, Body}}
```

Reply:
Hi, could you shortly explain why there should be "true = " before unalias, in this example?

Reply:
Sure. I had some doubts, that it's necessary, but to think about it this way, that will be added value to the code. Thanks for the quick reasoning.

Thanks for reviewing, 
BR,
Konrad